### PR TITLE
New multijob and phase variables

### PR DESCRIPTION
**Pull request by rrialq**
With the current multijob plugin we can not write conditions that includes:
* The 'Build Only If SCM Changes' value.
* The number of aborted jobs in previous phase.
* The number of stable jobs triggered in previous phase.
* The number of stable jobs triggered till now.
* And so on.

With this changes we can write conditions that includes the new variables:
* ${JOB_IS_BUILDABLE}
* ${PHASEJOB_SUCCESSFUL} > 4 || ${JOB_IS_BUILDABLE}
* ${MULTIJOB_ABORTED} == 0
* "${PHASEJOB_RESULT}" == "UNSTABLE" && ${JOB_IS_BUILDABLE} && ${MULTIJOB_UNSTABLE} == 0
* "${JOB_STATUS}" == "WORKSPACE_IS_EMPTY"

You can read the AboutNewVariables.txt and OtherImprovements.txt documents for more information.
This 
I hope it's helpful!

### CODE REVIEW

**Comment by rrialq** [comment]
hagzag  : This work is finished.
It can be merged with jenkinsci/tikal-multijob-plugin.

**Comment by rrialq** [comment]
@hagzag : This work is finished.
It can be merged with jenkinsci/tikal-multijob-plugin.

**Comment by harcher81** [comment]
Good job, this is very usefull.

**Comment by rrialq** [comment]
I've just discovered a bug into this branch.
Sometimes the MULTIJOB variables are showing incorrect values, so I must fix it before merging.
Sorry.

**Comment by rrialq** [comment]
Sorry. We had not detected this problem until today .
I uploaded the patch, and now works correctly .
We continue with testing.

**Comment by hagzag** [comment]
@cohencil Are you on this PR ?

**Comment by cohencil** [comment]
no

**Comment by rrialq** [comment]
@hagzag My tests are finished now.
No more problems detected since last commit.
This is a finished work, so if you consider useful, you can merge it.

File: src/main/java/com/tikal/jenkins/plugins/multijob/MultiJobBuilder.java [hunk 1]
```diff
@@ -591,14 +658,13 @@ private void addBuildEnvironmentVariables(MultiJobBuild thisBuild,
             variables.put("TRIGGERED_JOB_NAMES", triggeredJobNames);
         }
 
-        if (variables.get("TRIGGERED_BUILD_RUN_COUNT_" + jobNameSafe) == null) {
-            variables.put("TRIGGERED_BUILD_RUN_COUNT_" + jobNameSafe, "1");
+        if (variables.get("TRIGGERED_BUILD_RUN_COUNTER_" + jobNameSafe) == null) {
```
Comment (oleg-nenashev) [inline src/main/java/com/tikal/jenkins/plugins/multijob/MultiJobBuilder.java]: Seems it breaks the jobs, which rely on original variable names

File: AboutNewVariables.txt [hunk 2]
```diff
@@ -0,0 +1,97 @@
+About the new multijob, last phase, phase and job variables
```
Comment (oleg-nenashev) [inline AboutNewVariables.txt]: It should a *.md file with an integration with README.md

File: src/main/resources/com/tikal/jenkins/plugins/multijob/views/JobColumn/column.jelly [hunk 3]
```diff
@@ -7,7 +7,7 @@
 				${job.isConditional() ? "[?]" : ""} ${job.displayName}
 			</j:when>
 			<j:when test="${!job.phase}">
-				<a href="${jobBaseUrl}${job.shortUrl}">${job.displayName}</a>
+			<a href="${rootURL}/${job.url}">${job.displayName}</a>
```
Comment (oleg-nenashev) [inline src/main/resources/com/tikal/jenkins/plugins/multijob/views/JobColumn/column.jelly]: This approach may be wrong in particular cases #72 provides a right solution

**Comment by rrialq** [comment]
@oleg-nenashev, I fix the issues.

**Comment by Memphiz** [comment]
Are you sure this works as intended? We updated the plugin and now it doesn't build any subjobs anymore ... it just prints:

Job status: [WIN-32] subjob has no changes since last build. 

I don't have the "Build only if SCM changes" enabled on the subjobs so i want it to build everytime. There are other reasons why we want to rebuild a multijob with the same SHA1 (for example building debug and release builds from the same SHA1). Is this a bug or did we forget something after updating the plugin? @martijnkaijser fyi ping

**Comment by Memphiz** [comment]
Also it looks like this change adds the need to have git installed on the jenkins server ... this wasn't needed before ... without it and multijob plugin 1.19 i get an error that git-ls remote -h or something like that failed ... this is something it tries to execute on the master node (normally our master node has no git access at all and it was never needed) ... seems this changes a lot of behavior ...

**Comment by MartijnKaijser** [comment]
We updated no 1.20 and it worked for 1 or two days and now jenkins was not even loading anymore
See http://paste.ubuntu.com/13442272/

**Pull request merged**